plugin/aws/profile.go

```diff
@@ -23,8 +24,9 @@ func newProfile(ctx context.Context, name string) (*profile, error) {
 	// Create the session. SharedConfigEnable tells AWS to load the profile
 	// config from the ~/.aws/credentials and ~/.aws/config files
 	session, err := session.NewSessionWithOptions(session.Options{
-		Profile:           profile.Name(),
-		SharedConfigState: session.SharedConfigEnable,
+		Profile:                 profile.Name(),
+		AssumeRoleTokenProvider: stscreds.StdinTokenProvider,
```

Comment:
How long does the session created w/ `AssumeRoleTokenProvider` last? One thing I'm concerned about is that you can't use Wash's API with this unless you explicitly provide the MFA token via Stdin, so if you do something like `wash ls` inside the root, wouldn't Wash hang?

Reply:
Nice to see that it's very easy to add the MFA stuff though

Reply:
I'm not sure how long the session lasts. For the CLI it'll re-use a cached one for 6 hours. I'll see if I can find a documented answer, it wouldn't be great to have wash stop working after a couple of hours and have to restart it.

The experience of using this is that the wash server starts up normally, but the first time you list the `aws` plugin the wash server will prompt for an MFA token. My preference would be to force that to happen during server startup.

Reply:
It looks like the duration of the MFA authorization is specified as part of IAM policy. So we'll need to iterate on how re-authorization should work.

Reply:
Hmm, why not have this code be part of the `Init` method then? Seems like the user should be prompted for the MFA token when loading the AWS plugin.

We could handle re-authorization later as long as the session lasts long enough for us to do anything meaningful. That could be a known issue until that point, since it looks like we're going to be needing the MFA stuff in order to do anything useful with the AWS plugin.

Reply:
Ok, authorization is actually only requested when we need it, like when making a query to list S3 resources. So have to go pretty deep to trigger it.